editor/plugins/particle_process_material_editor_plugin.h

```diff
@@ -0,0 +1,138 @@
+/**************************************************************************/
+/*  particle_process_material_editor_plugin.h                             */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef PARTICLE_PROCESS_MATERIAL_EDITOR_PLUGIN_H
+#define PARTICLE_PROCESS_MATERIAL_EDITOR_PLUGIN_H
+
+#include "editor/editor_plugin.h"
+#include "editor/editor_properties.h"
+
+class Button;
+class EditorSpinSlider;
+class Label;
+class ParticleProcessMaterial;
+class Range;
+class VBoxContainer;
+
+class ParticleProcessMaterialMinMaxPropertyEditor : public EditorProperty {
+	GDCLASS(ParticleProcessMaterialMinMaxPropertyEditor, EditorProperty);
+
+	enum {
+		HOVER_NONE,
+		HOVER_LEFT,
+		HOVER_RIGHT,
+		HOVER_MIDDLE,
+	};
```

Comment:
Not that important, but I'd suggest we move away from anonymous C-style enums where possible.
E.g.:
```suggestion
	enum class Hover{
		NONE,
		LEFT,
		RIGHT,
		MIDDLE,
	};
```
We already did this with the Input enums: #51684

Reply:
I meant using `enum class` (C++ 11) which adds stronger type safety because named C-style enums unfortunately can be assigned in any (wrong) way you want. Using enum class also allows dropping the prefix of every enum value (e.g. `HOVER_LEFT` -> `Hover::LEFT`) because they have their own scope.